File: front-end/src/domain/valueobject/operationType.ts
import { Utils } from "../../utils/utils";

enum OperationType {
    LOAD,
    UNLOAD
}

class OperationTypeVO {
    private static OPERATIONTYPE_NULL_ERROR_MESSAGE: string = "The operation type cannot be null or empty";
    private static OPERATIONTYPE_INVALID_ERROR_MESSAGE: string = "The operation type specified does not exist";

    static validate(ordinal: number): string {
        let error: string = "";
        if (ordinal === null) {
            error = this.OPERATIONTYPE_NULL_ERROR_MESSAGE;
        }
        if (ordinal < 0 || ordinal >= Object.keys(OperationType).length / 2) {
            error = this.OPERATIONTYPE_INVALID_ERROR_MESSAGE;
        }
        return error;
    }

    static random(): OperationType {
        return Utils.randomInt(0, +Object.keys(OperationType) - 1);
    }

    static fromOrdinal(ordinal: number): String {
        return OperationType[ordinal];
    }
}

export { OperationType, OperationTypeVO };

File: back-end/src/main/java/com/trucks/domain/valueobject/OperationType.java
package com.trucks.domain.valueobject;

import java.util.Objects;
import java.util.Random;

public enum OperationType {

    LOAD(0),
    UNLOAD(1);

    private static OperationType[] values = null;

    static {
        values = OperationType.values();
    }

    private static final String OPERATIONTYPE_NULL_ERROR_MESSAGE =
            "The operation type cannot be null or empty";
    private static final String OPERATIONTYPE_INVALID_ERROR_MESSAGE =
            "The operation type specified does not exist";

    private OperationType(Integer ordinal) {
        String error = OperationType.validate(ordinal);
        if (!error.isEmpty()) {
            throw new IllegalArgumentException(error);
        }
    }

    public String toString() {
        return this.name();
    }

    public static String validate(Integer ordinal) {
        String error = "";
        if (ordinal == null)
            error = OPERATIONTYPE_NULL_ERROR_MESSAGE;
        if (ordinal < 0 || ordinal > 2)
            error = OPERATIONTYPE_INVALID_ERROR_MESSAGE;
        return error;
    }

    public static OperationType random() {
        Random rand = new Random();
        Integer randomNum = rand.nextInt(1);
        return values[randomNum];
    }

    public boolean equals(OperationType other) {
        return Objects.equals(this, other);
    }
}
